prediction_market_agent_tooling/tools/cowswap/encoding.py

```diff
@@ -0,0 +1,41 @@
+from eth_typing import ChecksumAddress
+from web3 import Web3
+
+from prediction_market_agent_tooling.gtypes import ChainID
+
+MESSAGE_TYPES_CANCELLATION = {
+    "OrderCancellations": [
+        {"name": "orderUids", "type": "bytes[]"},
+    ]
+}
+
+# EIP-712 Types
+MESSAGE_TYPES = {
+    "Order": [
+        {"name": "sellToken", "type": "address"},
+        {"name": "buyToken", "type": "address"},
+        {"name": "receiver", "type": "address"},
+        {"name": "sellAmount", "type": "uint256"},
+        {"name": "buyAmount", "type": "uint256"},
+        {"name": "validTo", "type": "uint32"},
+        {"name": "appData", "type": "bytes32"},
+        {"name": "feeAmount", "type": "uint256"},
+        {"name": "kind", "type": "string"},
+        {"name": "partiallyFillable", "type": "bool"},
+        {"name": "sellTokenBalance", "type": "string"},
+        {"name": "buyTokenBalance", "type": "string"},
+    ]
+}
+
+# EIP-712 Domain
+DOMAIN = {
+    "name": "Gnosis Protocol",
+    "version": "v2",
+    "chainId": 100,  # Replace with actual chainId
```

Comment:
We have https://github.com/gnosis/prediction-market-agent-tooling/blob/6055ed2ef7f5e35378a16ba8ca167ddba7239f80/prediction_market_agent_tooling/tools/gnosis_rpc.py#L5, can be used here

```suggestion
    "chainId": GNOSIS_NETWORK_ID,  # Replace with actual chainId
```